rust/otap-dataflow/crates/telemetry/src/opentelemetry_client.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Opentelemetry SDK integration for telemetry collection and reporting as a client.
+
+use std::time::Duration;
+
+use opentelemetry::{KeyValue, global};
+use opentelemetry_sdk::{
+    Resource,
+    metrics::{PeriodicReader, SdkMeterProvider},
+};
+use otap_df_config::pipeline::{
+    MetricsPeriodicExporterConfig, MetricsReaderConfig, TelemetryConfig,
+};
+
+/// Configure the global OpenTelemetry SDK according to the provided telemetry config.
+pub fn configure_otel_sdk(config: &TelemetryConfig) {
+    let mut sdk_meter_builder = SdkMeterProvider::builder();
+
+    // TODO: Load from config
+    // In the meantime, only configure it if there is one console metric reader..
+
+    let metric_readers = &config.metrics.readers;
+    for reader in metric_readers {
+        match reader {
+            MetricsReaderConfig::Periodic(periodic_config) => {
+                let interval = &periodic_config.interval;
+                match &periodic_config.exporter {
+                    MetricsPeriodicExporterConfig::Console(_console_config) => {
+                        let exporter = opentelemetry_stdout::MetricExporter::default();
+                        let reader = PeriodicReader::builder(exporter)
+                            .with_interval(Duration::from_millis(*interval))
+                            .build();
+                        sdk_meter_builder = sdk_meter_builder.with_reader(reader);
+                    }
+                    _ => {
+                        // Ignore other exporters
+                    }
+                }
+            }
+            _ => {
+                // Ignore other readers
+            }
+        }
+    }
+
+    let resource_attributes = &config.resource;
+    if !resource_attributes.is_empty() {
+        let mut sdk_resource_builder = Resource::builder();
+        for (k, v) in resource_attributes.iter() {
+            sdk_resource_builder =
+                sdk_resource_builder.with_attribute(KeyValue::new(k.clone(), v.clone()));
+        }
+        let sdk_resource = sdk_resource_builder.build();
+        sdk_meter_builder = sdk_meter_builder.with_resource(sdk_resource);
+    }
+
+    let sdk_meter_provider = sdk_meter_builder.build();
+
+    global::set_meter_provider(sdk_meter_provider);
```

Comment:
if we need to flush/shutdown, need to hold on to a clone of the meter_provider.
https://github.com/open-telemetry/opentelemetry-rust/blob/main/examples/metrics-basic/src/main.rs#L20

Reply:
global is only necessary, if components want to obtain a meter by calling `global::meter`. The alternative is to get meter directly from the `meter_provider`.

In OTel Rust, it creates a single std::thread per provider, which does the periodic exporting part. (And as of today, it's not controllable which core that thread runs. We should be able to add such support in OTel Rust, if its required for Arrow telemetry needs)